appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/service/SendKinServiceBase.java

```diff
@@ -6,14 +6,18 @@
 import android.os.IBinder;
 import android.support.annotation.NonNull;
 
+import org.kinecosystem.appsdiscovery.repositories.KinTransferCallback;
+
 import java.math.BigDecimal;
 
 public abstract class SendKinServiceBase extends Service {
 
-    //Developer needs to implements
-    //these is called from background thread so can be called async
-    public abstract @NonNull
-    KinTransferComplete transferKin(@NonNull final String toAddress, final int amount, @NonNull final String memo) throws KinTransferException;
+    protected KinTransferCallback transferKinCallback = null;
+
+    public void cancelCallback() { transferKinCallback = null; }
+
+    public abstract
+    void transferKin(@NonNull final String toAddress, final int amount, @NonNull final String memo,@NonNull KinTransferCallback callback);
```

Comment:
we need a sync and async method. Should leave the original transferKin method as is for backward compatibility